metricflow-semantics/metricflow_semantics/model/semantics/linkable_spec_resolver.py

```diff
@@ -192,10 +192,15 @@ def __init__(
                 continue
             metric_reference = MetricReference(metric.name)
             linkable_element_set_for_metric = self.get_linkable_elements_for_metrics([metric_reference])
+            defined_from_semantic_models = tuple(
+                self._semantic_model_lookup.get_semantic_model_for_measure(input_measure.measure_reference).reference
+                for input_measure in metric.input_measures
+            )
```

Comment:
Not so much this specific block, but how do we handle this type of resolution for query-level metrics? It feels like this has to be duplicating that work somehow, or that this, at least, could be done from an extension of the API that provides the saved query semantic model inputs. Or is this what that logic relies on for getting those semantic model references?

Anyway, I don't think that changes anything for this PR.